src/dsp_tools/commands/xmllib/models/resource.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+import warnings
+from collections import defaultdict
+from dataclasses import dataclass
+from dataclasses import field
+
+from lxml import etree
+
+from dsp_tools.commands.xmllib.models.values import IntValue
+from dsp_tools.commands.xmllib.models.values import LinkValue
+from dsp_tools.commands.xmllib.models.values import SimpleText
+from dsp_tools.commands.xmllib.models.values import Value
+from dsp_tools.commands.xmllib.utils import is_string
+from dsp_tools.models.custom_warnings import DspToolsUserWarning
+from dsp_tools.models.exceptions import InputError
+from dsp_tools.utils.uri_util import is_iiif_uri
+
+XML_NAMESPACE_MAP = {None: "https://dasch.swiss/schema", "xsi": "http://www.w3.org/2001/XMLSchema-instance"}
+DASCH_SCHEMA = "{https://dasch.swiss/schema}"
+
+
+@dataclass
+class ResourceCollection:
+    resources: dict[str, Resource] = field(default_factory=dict)
+
+    def to_list(self) -> list[Resource]:
+        return list(self.resources.values())
+
+    def with_Resource(self, resource: Resource) -> ResourceCollection:
+        self._raise_exists(resource.res_id)
+        self.resources[resource.res_id] = resource
+        return self
+
+    def with_EmptyResource(self, res_id: str, restype: str, label: str) -> ResourceCollection:
+        self._raise_exists(res_id)
+        self.resources[res_id] = Resource(res_id=res_id, restype=restype, label=label)
+        return self
+
+    def add_ValueToResource(self, res_id: str, value: Value) -> ResourceCollection:
+        self._raise_does_not_exist(res_id)
+        self.resources[res_id].values.append(value)
+        return self
+
+    def add_FileValueToResource(self, res_id: str, filepath: str) -> ResourceCollection:
+        self._raise_does_not_exist(res_id)
+        self.resources[res_id].filepath = filepath
+        return self
```

Comment:
I wouldn't expose this functionality - when something is added, it should not be modified anymore

Reply:
I understand the concern with modification. We can just add a class for the FileValue. But I think this functionality is particularly useful, because it allows you to add files later when all the resources were created.

Reply:
Do I understand your comment that it should not be modified, in the sense that you would just add values to a list with the resource id and then combine them only when you serialise it but not actually append it to the resource?

Reply:
no, my point is that no resource should end in a resource collection when it's not yet finished.